lib/notesclub_web/live/notesLive.ex

```diff
@@ -0,0 +1,107 @@
+defmodule NotesclubWeb.NotesLive do
+  use NotesclubWeb, :live_view
+  use Phoenix.HTML
+  import Phoenix.LiveView.Helpers
+  import Phoenix.LiveView
+
+  alias NotesclubWeb.NotesLive
+  alias Notesclub.Notebooks
+
+  @notebooks_in_home_count 7
+  def notebooks_in_home_count, do: @notebooks_in_home_count
+
+  def mount(_params, _session, socket) do
+    {:ok, socket}
+  end
+
+  def handle_event("search", %{"term" => term}, socket) do
+    {:noreply, push_patch(socket, to: Routes.notes_path(socket, :all, search: term))}
+  end
+
+  def handle_event("random", _, socket) do
+    {:noreply, push_patch(socket, to: Routes.live_path(socket, NotesLive))}
+  end
+
+  def handle_params(%{"author" => author, "repo" => repo}, _url, socket) do
+    notebooks = Notebooks.list_repo_author_notebooks_desc(repo, author)
+    count = Notebooks.count()
+
+    {:noreply,
+     assign(socket, notebooks: notebooks, notebooks_count: count, search: nil, page: :repo)}
+  end
+
+  def handle_params(%{"author" => author}, _url, socket) do
+    notebooks = Notebooks.list_author_notebooks_desc(author)
+    count = Notebooks.count()
+
+    {:noreply,
+     assign(socket, notebooks: notebooks, notebooks_count: count, search: nil, page: :author)}
+  end
+
+  def handle_params(%{"search" => "content:" <> term}, _url, socket) do
+    notebooks = Notebooks.list_notebooks(github_filename: term)
+
+    notebooks2 =
+      Notebooks.list_notebooks(content: term, exclude_ids: notebooks |> Enum.map(& &1.id))
+
+    count = Notebooks.count()
+
+    {:noreply,
+     assign(socket,
+       notebooks: notebooks ++ notebooks2,
+       notebooks_count: count,
+       search: term,
+       page: :all
+     )}
+  end
+
+  def handle_params(%{"search" => term}, _url, socket) do
+    notebooks = Notebooks.list_notebooks(github_filename: term)
+    count = Notebooks.count()
+
+    {:noreply,
+     assign(socket, notebooks: notebooks, notebooks_count: count, search: term, page: :all)}
+  end
+
+  def handle_params(%{}, url, socket) do
+    cond do
+      String.ends_with?(url, "/last_week") ->
```

Comment:
@hectorperez if you can do it i'll appreciate because i can't do it today and tomorrow is a day off here in Brazil and probably i wont be online.

I can't wait to help to improve Notesclub. It was a pleasure!

Reply:
@FabriDamazio I can do it on Saturday and merge the PR.

The pleasure’s mine!

Reply:
If i can do before saturday i msg here

Reply:
@hectorperez just made the change, please review it =)

Reply:
Looks great!!